p2p/host/autorelay/relay_finder.go

```diff
@@ -178,20 +207,59 @@ func (rf *relayFinder) background(ctx context.Context) {
 // It garbage collects old entries, so that nodes doesn't overflow.
 // This makes sure that as soon as we need to find relay candidates, we have them available.
 func (rf *relayFinder) findNodes(ctx context.Context) {
+	peerChan := rf.peerSource(rf.conf.maxCandidates)
```

Comment:
Don't you have to check if this func var may be nil? Or if you assume this is non nil, should we assert that in the constructor?

Reply:
That would be a violation of the API contract. Not sure if we have to defend against that (panic?).

`findNodes` is called repeatedly, so we can't move it to the constructor.